command/acl_auth_method_create.go

```diff
@@ -0,0 +1,174 @@
+	"github.com/posener/complete"
+	"golang.org/x/exp/slices"
+)
+
+// Ensure ACLRoleCreateCommand satisfies the cli.Command interface.
+var _ cli.Command = &ACLAuthMethodCreateCommand{}
+
+// ACLAuthMethodCreateCommand implements cli.Command.
+type ACLAuthMethodCreateCommand struct {
+	Meta
+
+	name          string
+	methodType    string
+	tokenLocality string
+	maxTokenTTL   time.Duration
+	isDefault     bool
+	config        string
+}
+
+// Help satisfies the cli.Command Help function.
+func (a *ACLAuthMethodCreateCommand) Help() string {
+	helpText := `
+Usage: nomad acl auth-method create [options]
+
+  Create is used to create new ACL auth methods. Use requires a management token.
+
+General Options:
+
+  ` + generalOptionsUsage(usageOptsDefault|usageOptsNoNamespace) + `
+
+ACL Auth Method Create Options:
+
+  -name
+    Sets the human readable name for the ACL auth method. The name must be
+    between 1-128 characters and is a required parameter.
+
+  -type
+    Sets the type of the auth method. Currently the only supported type is
+    'OIDC'.
+
+  -max-token-ttl
+    Sets the duration of time all tokens created by this auth method should be
+    valid for.
+
+  -token-locality
+    Defines the kind of token that this auth method should produce. This can be
+    either 'local' or 'global'. If empty the value of 'local' is assumed.
+
+  -default
+    Specifies whether this auth method should be treated as a default one in
+    case no auth method is explicitly specified for a login command.
+
+  -config
+    Auth method configuration in JSON format.
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (a *ACLAuthMethodCreateCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(a.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-name":           complete.PredictAnything,
+			"-type":           complete.PredictAnything,
+			"-max-token-ttl":  complete.PredictAnything,
+			"-token-locality": complete.PredictAnything,
+			"-default":        complete.PredictAnything,
+			"-config":         complete.PredictNothing,
+		})
+}
+
+func (a *ACLAuthMethodCreateCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+// Synopsis satisfies the cli.Command Synopsis function.
+func (a *ACLAuthMethodCreateCommand) Synopsis() string { return "Create a new ACL auth method" }
+
+// Name returns the name of this command.
+func (a *ACLAuthMethodCreateCommand) Name() string { return "acl auth-method create" }
+
+// Run satisfies the cli.Command Run function.
+func (a *ACLAuthMethodCreateCommand) Run(args []string) int {
+
+	flags := a.Meta.FlagSet(a.Name(), FlagSetClient)
+	flags.Usage = func() { a.Ui.Output(a.Help()) }
+	flags.StringVar(&a.name, "name", "", "")
+	flags.StringVar(&a.methodType, "type", "", "")
+	flags.StringVar(&a.tokenLocality, "token-locality", "local", "")
+	flags.DurationVar(&a.maxTokenTTL, "max-token-ttl", 0, "")
+	flags.BoolVar(&a.isDefault, "default", true, "")
```

Comment:
I wonder whether true is the right default here? I don't have the answer, just the curiosity.

Reply:
This is a general issue with default values for non-string flags. I don't know either, but I think we'd want at least one auth method to be the default, wouldn't we? Same problem for the TTL: I set it to 0 by default, but the validation function won't accept anything below min TTL set in the server config. 

Is there a clever way not to set default values here?

Reply:
The [helper/flags](https://github.com/hashicorp/nomad/blob/main/helper/flags/autopilot_flags.go) package has some helpers that allow the use of pointer values as flags. I wonder if this would helper, although its general use purpose is for updates where you merge two values.

I think defaulting to false is fine for this flag in general. My gut says that having an administrator have to set the method to be default, is better than assuming it will be the default.

The TTL has to be set by the administrator when using this command as we cannot predict (baring looking it up via an API) what the TTL min/max values are. Therefore doing a `if ttl < 1` seems fine to me.

Reply:
I made a change in 0a6a7a33929f75890264fb5fea023b77b47cdb0c, let me know if you find it ok.